plugins/ipam/dhcp/lease.go

```diff
@@ -344,6 +360,45 @@ func (l *DHCPLease) maintain() {
 	}
 }
 
+func (l *DHCPLease) checkLinkExistsWithBackoff(ctx context.Context) (bool, error) {
+	checkFunc := func() (bool, error) {
+		_, err := netlink.LinkByName(l.link.Attrs().Name)
+		if err != nil {
+			var linkNotFoundErr *netlink.LinkNotFoundError = &netlink.LinkNotFoundError{}
+			if errors.As(err, linkNotFoundErr) {
+				return false, nil
+			}
+			return false, err
+		}
+		return true, nil
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, linkCheckTotalTimeout)
+	defer cancel()
+
+	exists, err := backoffRetryLinkExists(ctx, linkCheckRetryMax, checkFunc)
+	return exists, err
+}
+
+func backoffRetryLinkExists(ctx context.Context, maxDelay time.Duration, f func() (bool, error)) (bool, error) {
+	baseDelay := linkCheckDelay0
+	for {
+		exists, err := f()
```

Comment:
Currently, this function is pretty for `backoffRetryLinkExists`, not for common retry, hence we may not required to have ` f func() (bool, error))` as argument, I guess. Instead of that, having `linkName` in function argument makes it simple, I feel.

```
func backoffRetryLinkExists(ctx context.Context, maxDelay time.Duration, linkName string) (bool, error) {
  checkFunc := func() (bool, error) {
        _, err := netlink.LinkByName(linkName)
        if err != nil {
            var linkNotFoundErr *netlink.LinkNotFoundError = &netlink.LinkNotFoundError{}
            if errors.As(err, linkNotFoundErr) {
                return false, nil
            }
            return false, err
        }
        return true, nil
    }
```